### PR TITLE
Fix engine-blobs hive tests

### PR DESCRIPTION
**Pull request by acolytec3**
Addresses a failing test noted in #2931 where we should be broadcasting the contents of our transaction pool to newly connected peers.  Still WIP

This PR is a continuation of #2931, to test hive and other notes: instructions are there.

This also fixes a performance issue in the `sendNewTransactionHashes`function in `txPool` where it was `awaiting` a devp2p message that never gets a response and was causing some devp2p message handlers to time out silently.

### CODE REVIEW

**Comment by holgerd77** [comment]
Actually that’s not only “a failing test” but real misbehavior in our client. 😋

**Comment by jochem-brouwer** [comment]
For docs: this is relevant and super nice to read https://github.com/ethereum/devp2p/blob/master/caps/eth.md#transaction-exchange

File: packages/client/src/service/fullethereumservice.ts [hunk 1]
```diff
@@ -143,6 +145,24 @@ export class FullEthereumService extends EthereumService {
...
+      // TODO: Should we do this if the txPool isn't started?
+      const txs: [number[], number[], Uint8Array[]] = [[], [], []]
+      for (const addr of this.txPool.pool) {
+        for (const tx of addr[1]) {
+          const rawTx = tx.tx
+          txs[0].push(rawTx.type)
+          if (rawTx.type !== 3) {
+            txs[1].push(rawTx.serialize().byteLength)
+          } else {
+            txs[1].push((rawTx as BlobEIP4844Transaction).serializeNetworkWrapper().byteLength)
```
Comment (g11tech) [inline packages/client/src/service/fullethereumservice.ts:159]: shouldn't we annouce tx hashes over there?

File: packages/client/src/service/service.ts [hunk 2]
```diff
@@ -77,9 +77,9 @@ export class Service {
     this.config.events.on(Event.POOL_PEER_BANNED, (peer) =>
       this.config.logger.debug(`Peer banned: ${peer}`)
     )
-    this.config.events.on(Event.POOL_PEER_ADDED, (peer) =>
+    this.config.events.on(Event.POOL_PEER_ADDED, (peer) => {
```
Comment (g11tech) [inline packages/client/src/service/service.ts:80]: i think the previous version was also fine

**Review by g11tech — APPROVED** [review]
lgtm :partying_face:

**Pull request merged**